crates/blockifier/src/execution/syscalls/syscall_tests/emit_event.rs

```diff
@@ -44,35 +45,58 @@ fn positive_flow(test_contract: FeatureContract, expected_gas: u64) {
     );
 }
 
+#[test_case(FeatureContract::TestContract(CairoVersion::Native); "Native")]
 #[test_case(FeatureContract::TestContract(CairoVersion::Cairo1); "VM")]
 fn data_length_exceeds_limit(test_contract: FeatureContract) {
     let versioned_constants = VersionedConstants::create_for_testing();
 
     let max_event_data_length = versioned_constants.tx_event_limits.max_data_length;
     let data_too_long = vec![felt!(2_u16); max_event_data_length + 1];
-    let error = emit_events(test_contract, &N_EMITTED_EVENTS, &KEYS, &data_too_long).unwrap_err();
+
+    let call_result = emit_events(test_contract, &N_EMITTED_EVENTS, &KEYS, &data_too_long);
+    let error_message =
+        if matches!(test_contract, FeatureContract::TestContract(CairoVersion::Native)) {
+            let call_info = call_result
+                .expect("Expected CallResult with failed execution error message in the retdata.");
+            decode_felts_as_str(&call_info.execution.retdata.0)
+        } else {
+            let error = call_result.unwrap_err();
+            error.to_string()
+        };
     let expected_error = EmitEventError::ExceedsMaxDataLength {
         data_length: max_event_data_length + 1,
         max_data_length: max_event_data_length,
     };
-    assert!(error.to_string().contains(&expected_error.to_string()));
+    assert!(error_message.contains(&expected_error.to_string()));
 }
 
+#[test_case(FeatureContract::TestContract(CairoVersion::Native); "Native")]
 #[test_case(FeatureContract::TestContract(CairoVersion::Cairo1); "VM")]
 fn keys_length_exceeds_limit(test_contract: FeatureContract) {
     let versioned_constants = VersionedConstants::create_for_testing();
 
     let max_event_keys_length = versioned_constants.tx_event_limits.max_keys_length;
     let keys_too_long = vec![felt!(1_u16); max_event_keys_length + 1];
-    let error = emit_events(test_contract, &N_EMITTED_EVENTS, &keys_too_long, &DATA).unwrap_err();
+
+    let call_result = emit_events(test_contract, &N_EMITTED_EVENTS, &keys_too_long, &DATA);
+    let error_message =
+        if matches!(test_contract, FeatureContract::TestContract(CairoVersion::Native)) {
+            let call_info = call_result
+                .expect("Expected CallResult with failed execution error message in the retdata.");
+            decode_felts_as_str(&call_info.execution.retdata.0)
+        } else {
+            let error = call_result.unwrap_err();
+            error.to_string()
+        };
```

Comment:
`emit_events` takes a `FeatureContract` as argument therefore I expect the way the error result is to be read out to be independent of the value of `FeatureContract`. This seems to point to incorrect error handling deeper down.

Reply:
This is actually just because it's not running native. For the Cairo1 test "VM", the native syscall handler for emit_event is never called.
<!-- Reviewable comment -OBRbNvWEHGlm2zzKj3n:bx1z39x -->
<!-- Sent from Reviewable.io -->